stats/stats.go

```diff
@@ -0,0 +1,54 @@
+// Licensed under the Apache License, Version 2.0
+// Details: https://raw.githubusercontent.com/maniksurtani/quotaservice/master/LICENSE
+
+package stats
+
+import (
+	"fmt"
+
+	"github.com/maniksurtani/quotaservice/events"
+)
+
+type Listener interface {
+	TopHits(string) []*BucketScore
+	TopMisses(string) []*BucketScore
+	Get(string, string) *BucketScores
+	HandleEvent(events.Event)
+}
+
+type BucketScores struct {
+	Hits   int64 `json:"hits"`
+	Misses int64 `json:"misses"`
+}
+
+type BucketScore struct {
+	Bucket string `json:"bucket"`
+	Score  int64  `json:"value"`
+}
+
+var emptyArr []*BucketScore
+var emptyBucketScores *BucketScores
+
+func init() {
+	emptyArr = make([]*BucketScore, 0)
+	emptyBucketScores = &BucketScores{0, 0}
+}
+
+func (b *BucketScore) String() string {
+	return fmt.Sprintf("{%s, %d}", b.Bucket, b.Score)
+}
+
+// Implements a sortable BucketScore array
+type BucketScoreArray []*BucketScore
+
+func (b BucketScoreArray) Len() int {
```

Comment:
Should these functions be on a `*BucketScoreArray` instead of a `BucketScoreArray`, to prevent a copy?